drivers/edac/edac_ibecc.c

```diff
@@ -0,0 +1,457 @@
+
+	LOG_DBG("Get Parity Error Log");
+
+	return sys_read64((mem_addr_t)((uint8_t *)data->mchbar +
+				       IBECC_PARITY_ERROR_LOG));
+}
+
+static void parity_error_log_clear(const struct device *dev)
+{
+	struct ibecc_data *data = dev->data;
+
+	LOG_DBG("Clearing Parity Error Log");
+
+	sys_write64(PARITY_ERROR_ERRSTS,
+		    (mm_reg_t)((uint8_t *)data->mchbar +
+			       IBECC_PARITY_ERROR_LOG));
+}
+
+static unsigned int errors_cor_get(const struct device *dev)
+{
+	struct ibecc_data *data = dev->data;
+
+	return data->errors_cor;
+}
+
+static unsigned int errors_uc_get(const struct device *dev)
+{
+	struct ibecc_data *data = dev->data;
+
+	return data->errors_uc;
+}
+
+static int notify_callback_set(const struct device *dev,
+			       edac_notify_callback_f cb)
+{
+	struct ibecc_data *data = dev->data;
+	int key = irq_lock();
+
+	data->cb = cb;
+	irq_unlock(key);
+
+	return 0;
+}
+
+static const struct edac_driver_api api = {
+#if defined(CONFIG_EDAC_ERROR_INJECT)
+	/* Error Injection functions */
+	.inject_addr_set = inject_addr_set,
+	.inject_addr_get = inject_addr_get,
+	.inject_addr_mask_set = inject_addr_mask_set,
+	.inject_addr_mask_get = inject_addr_mask_get,
+	.inject_ctrl_set = inject_ctrl_set,
+#endif /* CONFIG_EDAC_ERROR_INJECT */
+
+	/* Error reporting & clearing functions */
+	.ecc_error_log_get = ecc_error_log_get,
+	.ecc_error_log_clear = ecc_error_log_clear,
+	.parity_error_log_get = parity_error_log_get,
+	.parity_error_log_clear = parity_error_log_clear,
+
+	/* Get error stats */
+	.errors_cor_get = errors_cor_get,
+	.errors_uc_get = errors_uc_get,
+
+	/* Notification callback set */
+	.notify_cb_set = notify_callback_set,
+};
+
+int edac_ibecc_init(const struct device *dev)
+{
+	const pcie_bdf_t bdf = PCIE_BDF(0, 0, 0);
+	struct ibecc_data *data = dev->data;
+	uint32_t tolud;
+	uint64_t touud, tom, mchbar;
+
+	ARG_UNUSED(dev);
+
+	LOG_INF("EDAC IBECC initialization");
+
+	if (!pcie_probe(bdf, PCIE_ID(PCI_VENDOR_ID_INTEL,
+				     PCI_DEVICE_ID_SKU7)) &&
+	    !pcie_probe(bdf, PCIE_ID(PCI_VENDOR_ID_INTEL,
+				     PCI_DEVICE_ID_SKU12))) {
+		LOG_ERR("Probe failed");
+		return -ENODEV;
+	}
+
+	if (!ibecc_enabled(bdf)) {
+		LOG_ERR("IBECC is not enabled");
+		return -ENODEV;
+	}
+
+	mchbar = pcie_conf_read(bdf, MCHBAR_REG);
+	mchbar |= (uint64_t)pcie_conf_read(bdf, MCHBAR_REG + 1) << 32;
+
+	/* Check that MCHBAR is enabled */
+	if (!(mchbar & MCHBAR_ENABLE)) {
+		LOG_ERR("MCHBAR is not enabled");
+		return -ENODEV;
+	}
+
+	mchbar &= GENMASK(38, 16);
+
+	/* workaround for 32 bit read */
+	touud = pcie_conf_read(bdf, TOUUD_REG);
+	touud |= (uint64_t)pcie_conf_read(bdf, TOUUD_REG + 1) << 32;
+	touud &= GENMASK(38, 20);
+
+	/* workaround for 32 bit read */
+	tom = pcie_conf_read(bdf, TOM_REG);
+	tom |= (uint64_t)pcie_conf_read(bdf, TOM_REG + 1) << 32;
+	tom &= GENMASK(38, 20);
+
+	tolud = pcie_conf_read(bdf, TOLUD_REG) & GENMASK(31, 20);
+
+	device_map(&data->mchbar, mchbar, 0x10000, K_MEM_CACHE_NONE);
```

Comment:
What is the reason for not using pcie_get_mbar() ? 
The value 0x10000 comes from the datasheet? (pcie_get_mbar() would give you that info btw).

Reply:
MCHBAR is base address for the Host Memory Mapped Configuration space, at 0x48.

Reply:
You should avoid magic numbers and declare this with details about where this number came from.

Reply:
I will fix this